sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/DefaultServiceSpecBuilder.java

```diff
@@ -0,0 +1,82 @@
+package com.mesosphere.sdk.specification.yaml;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.commons.io.FileUtils;
+
+import com.mesosphere.sdk.config.DefaultTaskEnvRouter;
+import com.mesosphere.sdk.scheduler.SchedulerFlags;
+import com.mesosphere.sdk.specification.DefaultServiceSpec;
+import com.mesosphere.sdk.specification.ServiceSpec;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Generates a {@link ServiceSpec} from a given YAML definition in the form of a {@link RawServiceSpec}.
+ */
+public class DefaultServiceSpecBuilder {
+
+    /**
+     * Implementation for reading files from disk. Meant to be overridden by a mock in tests.
+     */
+    @VisibleForTesting
+    public static class FileReader {
+        public String read(String path) throws IOException {
+            return FileUtils.readFileToString(new File(path), StandardCharsets.UTF_8);
```

Comment:
I was curious if there was a Java 8 native method for this. There isn't precisely (doesn't let you specify a CHARSET). Boo 👎